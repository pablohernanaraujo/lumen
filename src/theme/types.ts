import type { ImageStyle, TextStyle, ViewStyle } from 'react-native';

import type { colorTokens } from './colors';
import type { baseTokens } from './tokens';

export type Mode = keyof typeof colorTokens;

export type BaseTokens = typeof baseTokens;

export type ColorTokens = typeof colorTokens;

export type ThemeColors = (typeof colorTokens)[keyof typeof colorTokens];

export type Theme = BaseTokens & {
  colors: ThemeColors;
  mode: Mode;
};

export type StyleFunction<T> = (theme: Theme) => T;

export type NamedStyles<T> = {
  [P in keyof T]: ViewStyle | TextStyle | ImageStyle;
};

export type StylesOrFunction<T extends NamedStyles<T>> = T | StyleFunction<T>;

export interface ThemeContextValue {
  theme: Theme;
  mode: Mode;
  setMode: (mode: Mode) => void;
  toggleMode: () => void;
}

export interface ThemeProviderProps {
  children: React.ReactNode;
  preference?: Mode;
  persistenceKey?: string;
}
