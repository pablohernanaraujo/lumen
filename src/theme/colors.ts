export const colorTokens = {
  light: {
    background: '#FFFFFF',
    surface: '#F8F9FA',
    surfaceSecondary: '#F1F3F4',
    text: {
      primary: '#1A1A1A',
      secondary: '#6B7280',
      tertiary: '#9CA3AF',
      inverse: '#FFFFFF',
    },
    primary: {
      50: '#EFF6FF',
      100: '#DBEAFE',
      200: '#BFDBFE',
      300: '#93C5FD',
      400: '#60A5FA',
      500: '#3B82F6',
      600: '#2563EB',
      700: '#1D4ED8',
      800: '#1E40AF',
      900: '#1E3A8A',
      main: '#3B82F6',
      dark: '#1D4ED8',
      light: '#93C5FD',
    },
    secondary: {
      50: '#FFF7ED',
      100: '#FFEDD5',
      200: '#FED7AA',
      300: '#FDBA74',
      400: '#FB923C',
      500: '#F97316',
      600: '#EA580C',
      700: '#C2410C',
      800: '#9A3412',
      900: '#7C2D12',
      main: '#F97316',
      dark: '#C2410C',
      light: '#FDBA74',
    },
    success: {
      main: '#10B981',
      dark: '#059669',
      light: '#6EE7B7',
    },
    warning: {
      main: '#F59E0B',
      dark: '#D97706',
      light: '#FCD34D',
    },
    error: {
      main: '#EF4444',
      dark: '#DC2626',
      light: '#FCA5A5',
    },
    info: {
      main: '#3B82F6',
      dark: '#1D4ED8',
      light: '#93C5FD',
    },
    border: '#E5E7EB',
    borderLight: '#F3F4F6',
    borderDark: '#D1D5DB',
    overlay: 'rgba(0, 0, 0, 0.5)',
  },
  dark: {
    background: '#000000',
    surface: '#111111',
    surfaceSecondary: '#1F1F1F',
    text: {
      primary: '#FFFFFF',
      secondary: '#A1A1AA',
      tertiary: '#71717A',
      inverse: '#000000',
    },
    primary: {
      50: '#1E3A8A',
      100: '#1E40AF',
      200: '#1D4ED8',
      300: '#2563EB',
      400: '#3B82F6',
      500: '#60A5FA',
      600: '#93C5FD',
      700: '#BFDBFE',
      800: '#DBEAFE',
      900: '#EFF6FF',
      main: '#60A5FA',
      dark: '#2563EB',
      light: '#BFDBFE',
    },
    secondary: {
      50: '#7C2D12',
      100: '#9A3412',
      200: '#C2410C',
      300: '#EA580C',
      400: '#F97316',
      500: '#FB923C',
      600: '#FDBA74',
      700: '#FED7AA',
      800: '#FFEDD5',
      900: '#FFF7ED',
      main: '#FB923C',
      dark: '#EA580C',
      light: '#FED7AA',
    },
    success: {
      main: '#34D399',
      dark: '#10B981',
      light: '#A7F3D0',
    },
    warning: {
      main: '#FBBF24',
      dark: '#F59E0B',
      light: '#FEF3C7',
    },
    error: {
      main: '#F87171',
      dark: '#EF4444',
      light: '#FECACA',
    },
    info: {
      main: '#60A5FA',
      dark: '#3B82F6',
      light: '#BFDBFE',
    },
    border: '#374151',
    borderLight: '#4B5563',
    borderDark: '#1F2937',
    overlay: 'rgba(0, 0, 0, 0.8)',
  },
} as const;
