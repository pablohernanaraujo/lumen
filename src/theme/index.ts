export { colorTokens } from './colors';
export { makeStyles } from './make-styles';
export { ThemeProvider, useTheme } from './theme-provider';
export { baseTokens } from './tokens';
export type {
  BaseTokens,
  ColorTokens,
  EmphasisLevel,
  Mode,
  NamedStyles,
  StyleFunction,
  StylesOrFunction,
  Theme,
  ThemeColors,
  ThemeContextValue,
  ThemeProviderProps,
} from './types';
export {
  createButtonStyle,
  createCardStyle,
  createInputStyle,
  createShadow,
  createTextStyle,
  getColorOpacity,
  getContrastColor,
  isLightColor,
} from './utils';
