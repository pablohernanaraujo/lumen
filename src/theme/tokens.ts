export const baseTokens = {
  spacing: {
    none: 0,
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 48,
  },
  radii: {
    sm: 4,
    md: 8,
    lg: 16,
    xl: 24,
    round: 999,
  },
  typography: {
    family: {
      light: 'Nunito-Light',
      regular: 'Nunito-Regular',
      medium: 'Nunito-Medium',
      semibold: 'Nunito-SemiBold',
      bold: 'Nunito-Bold',
      mono: 'Courier',
    },
    size: {
      xs: 12,
      sm: 14,
      md: 16,
      lg: 18,
      xl: 20,
      xxl: 24,
      xxxl: 32,
    },
    weight: {
      light: '300' as const,
      regular: '400' as const,
      medium: '500' as const,
      semibold: '600' as const,
      bold: '700' as const,
    },
    lineHeight: {
      tight: 1.2,
      normal: 1.4,
      relaxed: 1.6,
    },
  },
  shadows: {
    sm: {
      shadowColor: '#000000',
      shadowOffset: {
        width: 0,
        height: 1,
      },
      shadowOpacity: 0.05,
      shadowRadius: 2,
      elevation: 2,
    },
    md: {
      shadowColor: '#000000',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.1,
      shadowRadius: 4,
      elevation: 4,
    },
    lg: {
      shadowColor: '#000000',
      shadowOffset: {
        width: 0,
        height: 4,
      },
      shadowOpacity: 0.15,
      shadowRadius: 8,
      elevation: 8,
    },
  },
  opacity: {
    disabled: 0.6,
    overlay: 0.8,
    subtle: 0.4,
  },
  emphasis: {
    pure: 1,
    high: 0.87,
    medium: 0.6,
    low: 0.38,
  },
} as const;
