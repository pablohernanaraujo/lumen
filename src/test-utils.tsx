import React, { type FC, type ReactElement } from 'react';
import {
  render,
  type RenderOptions,
  type RenderResult,
} from '@testing-library/react-native';

import { ThemeProvider } from './theme';
import type { Mode } from './theme/types';

interface AllTheProvidersProps {
  children: React.ReactNode;
  mode?: Mode;
}

const AllTheProviders: FC<AllTheProvidersProps> = ({
  children,
  mode = 'light',
}) => <ThemeProvider preference={mode}>{children}</ThemeProvider>;

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  mode?: Mode;
  wrapper?: React.ComponentType<{ children: React.ReactNode }>;
}

export const customRender = (
  ui: ReactElement,
  { mode = 'light', wrapper, ...renderOptions }: CustomRenderOptions = {},
): RenderResult => {
  const Wrapper: FC<{ children: React.ReactNode }> = ({ children }) => {
    const content = <AllTheProviders mode={mode}>{children}</AllTheProviders>;

    return wrapper
      ? React.createElement(wrapper, { children: content })
      : content;
  };

  return render(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
};

// Re-export everything from React Native Testing Library
export * from '@testing-library/react-native';

// Override the default render with our custom one
export { customRender as render };

// Create snapshot matcher helper
export const createSnapshot = (
  component: ReactElement,
  mode: Mode = 'light',
): void => {
  const { toJSON } = customRender(component, { mode });
  expect(toJSON()).toMatchSnapshot();
};

// Create both light and dark mode snapshots
export const createDualSnapshots = (component: ReactElement): void => {
  describe('snapshots', () => {
    it('should render correctly in light mode', () => {
      createSnapshot(component, 'light');
    });

    it('should render correctly in dark mode', () => {
      createSnapshot(component, 'dark');
    });
  });
};

// Export TestWrapper for use in tests
export const TestWrapper: FC<{ children: React.ReactNode; mode?: Mode }> = ({
  children,
  mode = 'light',
}) => <AllTheProviders mode={mode}>{children}</AllTheProviders>;
