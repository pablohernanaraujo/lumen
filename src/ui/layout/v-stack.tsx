import React, {
  Children,
  cloneElement,
  type FC,
  isValidElement,
  ReactElement,
} from 'react';
import { View } from 'react-native';

import { makeStyles, useTheme } from '../../theme';
import type { VStackProps } from './types';

const useStyles = makeStyles(() => ({
  container: {
    flexDirection: 'column',
  },
}));

export const VStack: FC<VStackProps> = ({
  children,
  spacing = 'md',
  style,
}) => {
  const styles = useStyles();
  const { theme } = useTheme();

  const spacingValue = theme.spacing[spacing];

  const childrenArray = Children.toArray(children);

  return (
    <View style={[styles.container, style]}>
      {childrenArray.map((child, index) => {
        const isLastChild = index === childrenArray.length - 1;

        if (isValidElement(child)) {
          return cloneElement(child as ReactElement<{ style?: object }>, {
            key: child.key || index,
            style: [
              (child.props as { style?: object }).style,
              !isLastChild && { marginBottom: spacingValue },
            ],
          });
        }

        return (
          <View
            key={index}
            style={!isLastChild ? { marginBottom: spacingValue } : undefined}
          >
            {child}
          </View>
        );
      })}
    </View>
  );
};
