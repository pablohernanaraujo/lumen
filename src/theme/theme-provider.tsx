import React, {
  createContext,
  type FC,
  useCallback,
  useContext,
  useMemo,
  useState,
} from 'react';
import { useColorScheme } from 'react-native';

import { colorTokens } from './colors';
import { baseTokens } from './tokens';
import type {
  Mode,
  Theme,
  ThemeContextValue,
  ThemeProviderProps,
} from './types';

const ThemeContext = createContext<ThemeContextValue | null>(null);

export const ThemeProvider: FC<ThemeProviderProps> = ({
  children,
  preference,
  persistenceKey = '@lumen-theme',
}) => {
  const systemMode = useColorScheme() as Mode | null;
  const [userMode, setUserMode] = useState<Mode | null>(preference || null);

  const currentMode: Mode = userMode ?? systemMode ?? 'light';

  const theme = useMemo<Theme>(
    () => ({
      ...baseTokens,
      colors: colorTokens[currentMode],
      mode: currentMode,
    }),
    [currentMode],
  );

  const setMode = useCallback((mode: Mode): void => {
    setUserMode(mode);
  }, []);

  const toggleMode = useCallback((): void => {
    setUserMode(currentMode === 'light' ? 'dark' : 'light');
  }, [currentMode]);

  const contextValue = useMemo<ThemeContextValue>(
    () => ({
      theme,
      mode: currentMode,
      setMode,
      toggleMode,
    }),
    [theme, currentMode, setMode, toggleMode],
  );

  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = (): ThemeContextValue => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};
