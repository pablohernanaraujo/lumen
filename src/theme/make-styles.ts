import { StyleSheet } from 'react-native';

import { useTheme } from './theme-provider';
import type { NamedStyles, StylesOrFunction, Theme } from './types';

export const makeStyles =
  <T extends NamedStyles<T>>(
    factory: StylesOrFunction<T>,
  ): (() => StyleSheet.NamedStyles<T>) =>
  (): StyleSheet.NamedStyles<T> => {
    const { theme } = useTheme();

    const styles =
      typeof factory === 'function'
        ? (factory as (theme: Theme) => T)(theme)
        : factory;

    return StyleSheet.create(styles);
  };
