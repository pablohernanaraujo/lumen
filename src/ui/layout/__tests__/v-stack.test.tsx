import React, { FC } from 'react';
import { Text, View } from 'react-native';

import { createDualSnapshots, render } from '../../../test-utils';
import { makeStyles } from '../../../theme';
import { VStack } from '../v-stack';

const useTestStyles = makeStyles((theme) => ({
  redContainer: {
    backgroundColor: theme.colors.error.main,
    padding: theme.spacing.sm,
  },
  blueContainer: {
    backgroundColor: theme.colors.info.main,
  },
  smallRedBox: {
    width: 20,
    height: 20,
    backgroundColor: theme.colors.error.main,
  },
  redText: {
    color: theme.colors.error.main,
  },
  boldText: {
    fontWeight: theme.typography.weight.bold,
  },
  bluePaddedContainer: {
    backgroundColor: theme.colors.info.main,
    padding: theme.spacing.xs,
  },
}));

describe('VStack', () => {
  const mockChildren = [
    <Text key="1">First</Text>,
    <Text key="2">Second</Text>,
    <Text key="3">Third</Text>,
  ];

  createDualSnapshots(<VStack>{mockChildren}</VStack>);

  describe('with different spacing', () => {
    const spacingValues = ['xs', 'sm', 'md', 'lg', 'xl'] as const;

    for (const spacing of spacingValues) {
      it(`should render correctly with ${spacing} spacing`, () => {
        const { toJSON } = render(
          <VStack spacing={spacing}>{mockChildren}</VStack>,
        );
        expect(toJSON()).toMatchSnapshot();
      });
    }
  });

  describe('with custom styling', () => {
    it('should render correctly with custom style', () => {
      const TestComponent: FC = () => {
        const testStyles = useTestStyles();
        return <VStack style={testStyles.redContainer}>{mockChildren}</VStack>;
      };
      const { toJSON } = render(<TestComponent />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with custom style and spacing', () => {
      const TestComponent: FC = () => {
        const testStyles = useTestStyles();
        return (
          <VStack spacing="lg" style={testStyles.blueContainer}>
            {mockChildren}
          </VStack>
        );
      };
      const { toJSON } = render(<TestComponent />);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('with different children types', () => {
    it('should render correctly with single child', () => {
      const { toJSON } = render(
        <VStack>
          <Text>Single Child</Text>
        </VStack>,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with mixed children types', () => {
      const TestComponent: FC = () => {
        const testStyles = useTestStyles();
        return (
          <VStack>
            <Text>Text Child</Text>
            <View style={testStyles.smallRedBox} />
            <Text>Another Text</Text>
          </VStack>
        );
      };
      const { toJSON } = render(<TestComponent />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with children having existing styles', () => {
      const TestComponent: FC = () => {
        const testStyles = useTestStyles();
        return (
          <VStack spacing="lg">
            <Text style={testStyles.redText}>Styled Text 1</Text>
            <Text style={testStyles.boldText}>Styled Text 2</Text>
            <View style={testStyles.bluePaddedContainer}>
              <Text>Nested Content</Text>
            </View>
          </VStack>
        );
      };
      const { toJSON } = render(<TestComponent />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with string children', () => {
      const { toJSON } = render(
        <VStack>
          {'String child 1'}
          {'String child 2'}
          {'String child 3'}
        </VStack>,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with null/undefined children', () => {
      const { toJSON } = render(
        <VStack>
          <Text>Valid Child</Text>
          {null}
          {undefined}
          <Text>Another Valid Child</Text>
        </VStack>,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('edge cases', () => {
    it('should render correctly with no children', () => {
      const { toJSON } = render(<VStack>{[]}</VStack>);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with empty children', () => {
      const { toJSON } = render(<VStack>{null}</VStack>);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with children without keys', () => {
      const { toJSON } = render(
        <VStack>
          <Text>No Key 1</Text>
          <Text>No Key 2</Text>
        </VStack>,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('spacing behavior', () => {
    it('should not apply margin to last child', () => {
      const { toJSON } = render(
        <VStack spacing="lg">
          <Text>First</Text>
          <Text>Last</Text>
        </VStack>,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('should apply correct spacing between multiple children', () => {
      const { toJSON } = render(
        <VStack spacing="xl">
          <Text>Child 1</Text>
          <Text>Child 2</Text>
          <Text>Child 3</Text>
          <Text>Child 4</Text>
        </VStack>,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
