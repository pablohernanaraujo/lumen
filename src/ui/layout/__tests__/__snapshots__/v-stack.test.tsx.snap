// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VStack edge cases should render correctly with children without keys 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    No Key 1
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    No Key 2
  </Text>
</View>
`;

exports[`VStack edge cases should render correctly with empty children 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;

exports[`VStack edge cases should render correctly with no children 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;

exports[`VStack snapshots should render correctly in dark mode 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;

exports[`VStack snapshots should render correctly in light mode 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;

exports[`VStack spacing behavior should apply correct spacing between multiple children 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 32,
        },
      ]
    }
  >
    Child 1
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 32,
        },
      ]
    }
  >
    Child 2
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 32,
        },
      ]
    }
  >
    Child 3
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Child 4
  </Text>
</View>
`;

exports[`VStack spacing behavior should not apply margin to last child 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Last
  </Text>
</View>
`;

exports[`VStack with custom styling should render correctly with custom style 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#EF4444",
        "padding": 8,
      },
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;

exports[`VStack with custom styling should render correctly with custom style and spacing 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#3B82F6",
      },
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;

exports[`VStack with different children types should render correctly with children having existing styles 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#EF4444",
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Styled Text 1
  </Text>
  <Text
    style={
      [
        {
          "fontWeight": "700",
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Styled Text 2
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#3B82F6",
          "padding": 4,
        },
        false,
      ]
    }
  >
    <Text>
      Nested Content
    </Text>
  </View>
</View>
`;

exports[`VStack with different children types should render correctly with mixed children types 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    Text Child
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#EF4444",
          "height": 20,
          "width": 20,
        },
        {
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Another Text
  </Text>
</View>
`;

exports[`VStack with different children types should render correctly with null/undefined children 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    Valid Child
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Another Valid Child
  </Text>
</View>
`;

exports[`VStack with different children types should render correctly with single child 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Single Child
  </Text>
</View>
`;

exports[`VStack with different children types should render correctly with string children 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    String child 1
  </View>
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    String child 2
  </View>
  <View>
    String child 3
  </View>
</View>
`;

exports[`VStack with different spacing should render correctly with lg spacing 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;

exports[`VStack with different spacing should render correctly with md spacing 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 16,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;

exports[`VStack with different spacing should render correctly with sm spacing 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 8,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 8,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;

exports[`VStack with different spacing should render correctly with xl spacing 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 32,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 32,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;

exports[`VStack with different spacing should render correctly with xs spacing 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 4,
        },
      ]
    }
  >
    First
  </Text>
  <Text
    style={
      [
        undefined,
        {
          "marginBottom": 4,
        },
      ]
    }
  >
    Second
  </Text>
  <Text
    style={
      [
        undefined,
        false,
      ]
    }
  >
    Third
  </Text>
</View>
`;
